contracts/JBController.sol

```diff
@@ -6,81 +6,74 @@ import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
-import {IJBSplitAllocator} from "./interfaces/IJBSplitAllocator.sol";
-import {IJBSplitsStore} from "./interfaces/IJBSplitsStore.sol";
-import {IJBTokenStore} from "./interfaces/IJBTokenStore.sol";
+import {IJBSplitHook} from "./interfaces/IJBSplitHook.sol";
+import {IJBSplits} from "./interfaces/IJBSplits.sol";
 import {IJBToken} from "./interfaces/IJBToken.sol";
-import {IJBPriceFeed} from "./interfaces/IJBPriceFeed.sol";
+import {IJBTokens} from "./interfaces/IJBTokens.sol";
 import {JBConstants} from "./libraries/JBConstants.sol";
-import {JBFundingCycleMetadataResolver} from "./libraries/JBFundingCycleMetadataResolver.sol";
-import {JBOperations} from "./libraries/JBOperations.sol";
-import {JBSplitsGroups} from "./libraries/JBSplitsGroups.sol";
-import {JBFundingCycle} from "./structs/JBFundingCycle.sol";
-import {JBFundingCycleConfig} from "./structs/JBFundingCycleConfig.sol";
-import {JBFundingCycleMetadata} from "./structs/JBFundingCycleMetadata.sol";
+import {JBRulesetMetadataResolver} from "./libraries/JBRulesetMetadataResolver.sol";
+import {JBPermissionIds} from "./libraries/JBPermissionIds.sol";
+import {JBSplitGroupIds} from "./libraries/JBSplitGroupIds.sol";
+import {JBRuleset} from "./structs/JBRuleset.sol";
+import {JBRulesetConfig} from "./structs/JBRulesetConfig.sol";
+import {JBRulesetMetadata} from "./structs/JBRulesetMetadata.sol";
 import {JBTerminalConfig} from "./structs/JBTerminalConfig.sol";
 import {JBSplit} from "./structs/JBSplit.sol";
-import {JBSplitAllocationData} from "./structs/JBSplitAllocationData.sol";
-import {JBGroupedSplits} from "./structs/JBGroupedSplits.sol";
-
-/// @notice Stitches together funding cycles and project tokens, making sure all activity is accounted for and correct.
-contract JBController3_1 is
-    JBOperatable,
-    ERC2771Context,
-    ERC165,
-    IJBController3_1,
-    IJBMigratable
-{
-    // A library that parses the packed funding cycle metadata into a more friendly format.
-    using JBFundingCycleMetadataResolver for JBFundingCycle;
+import {JBSplitGroup} from "./structs/JBSplitGroup.sol";
+import {JBSplitHookPayload} from "./structs/JBSplitHookPayload.sol";
+
+/// @notice Stitches together rulesets and project tokens, making sure all activity is accounted for and correct.
+contract JBController is JBPermissioned, ERC2771Context, ERC165, IJBController, IJBMigratable {
+    // A library that parses packed ruleset metadata into a friendlier format.
+    using JBRulesetMetadataResolver for JBRuleset;
 
     //*********************************************************************//
     // --------------------------- custom errors ------------------------- //
     //*********************************************************************//
 
-    error BURN_PAUSED_AND_SENDER_NOT_VALID_TERMINAL_DELEGATE();
-    error FUNDING_CYCLE_ALREADY_LAUNCHED();
+    error BURN_PAUSED_AND_SENDER_NOT_VALID_TERMINAL_HOOK();
```

Comment:
unused, coming later?

Reply:
nope, remove. we removed the pauseBurn metadata flag. good call